docs/ref/checks.txt

```diff
@@ -313,6 +313,7 @@ Models
   ``<model>``.
 * **models.E032**: constraint name ``<constraint>`` is not unique amongst
   models: ``<model list>``.
+* **models.E033**: index name is not valid ``<index>``.
```

Comment:
This doesn't match to checks added in `django/core/checks/model_checks.py`, we should add two checks:

- ``` models.E033: Index names cannot start with an underscore or a number: <index>.```,
- ``` models.E034: Index names cannot be longer than <max_length> characters: <index>.```